pkg/store/proxy.go

```diff
@@ -470,6 +490,14 @@ func startStreamSeriesSet(
 				}
 			}
 		}()
+
+		shardMatcher := shardInfo.Matcher()
+		applySharding := shardInfo != nil && !storeSupportsSharding
+		if applySharding {
```

Comment:
This makes sure we select the correct subset of series in the querier in case the underlying store does not support series sharding.

Reply:
Let's keep it for safeguard as during the migration some old stores might still there.